lib/ash_authentication/strategies/api_key/plug.ex

```diff
@@ -183,4 +190,14 @@ defmodule AshAuthentication.Strategy.ApiKey.Plug do
       context: context
     )
   end
+
+  defp validate_header_prefix(prefix) when is_binary(prefix), do: :ok
+
+  defp validate_header_prefix(%Regex{} = prefix) do
```

Comment:
The regex validation approach using string inspection is a bit fragile, but I can't think of a better way to do it either 🤷‍♂️ This works fine for the use case.